crates/radix-engine-toolkit/src/transaction_types/types.rs

```diff
@@ -373,6 +375,50 @@ impl<'r> TransactionTypesReceipt<'r> {
             )
             .collect()
     }
+
+    pub fn new_nonfungibles(&self) -> Vec<NonFungibleGlobalId> {
+        let mut minted_id_list = Vec::new();
+        let mut burnt_id_list = Vec::new();
+        for (event_type, event_payload) in
+            self.commit_result.application_events.iter()
+        {
+            match event_type.0 {
+                Emitter::Method(node_id, ..) => {
+                    if node_id.entity_type().unwrap()
+                        == EntityType::GlobalNonFungibleResourceManager
+                    {
+                        let address: ResourceAddress =
+                            node_id.try_into().unwrap();
+                        if event_type.1
+                            == MintNonFungibleResourceEvent::EVENT_NAME
+                        {
+                            let event: MintNonFungibleResourceEvent =
+                                scrypto_decode(event_payload).unwrap();
+                            for local_id in event.ids {
+                                minted_id_list.push(NonFungibleGlobalId::new(
+                                    address, local_id,
+                                ));
+                            }
+                        } else if event_type.1
+                            == BurnNonFungibleResourceEvent::EVENT_NAME
+                        {
+                            let event: BurnNonFungibleResourceEvent =
+                                scrypto_decode(event_payload).unwrap();
+                            for local_id in event.ids {
+                                burnt_id_list.push(NonFungibleGlobalId::new(
+                                    address, local_id,
+                                ));
+                            }
+                        }
+                    }
+                }
+                _ => (),
+            }
+        }
+
+        minted_id_list.retain(|item| !burnt_id_list.contains(item));
+        minted_id_list
```

Comment:
Can we return a `HashSet<NonFungibleGlobalId>` here and keep using `Vec` in uniffi?